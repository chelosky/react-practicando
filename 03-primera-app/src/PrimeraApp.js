import React from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo}) => {

    return (
        <>
            <h1>{saludo}</h1>
            {/* <h1>Hola Mundo</h1> */}
            <p>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Relleno'
}

export default PrimeraApp;