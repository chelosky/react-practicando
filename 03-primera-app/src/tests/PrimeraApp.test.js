import React from "react"
import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp"
import { shallow } from "enzyme"
import '@testing-library/jest-dom';


describe('Primera App', () => {
    // test('test', () => {
    //     const saludo = 'KALADIN';

    //     // const wrapper = render( <PrimeraApp saludo={saludo} />);
    //     // wrapper.getByText(saludo);
        
    //     const {getByText} = render( <PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('test', () => {
        const saludo = 'KALADIN';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('test 2', () => {
        const saludo = 'KALADIN';
        const subtitulo = 'Syl';
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo} 
            />
        );
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    })
})