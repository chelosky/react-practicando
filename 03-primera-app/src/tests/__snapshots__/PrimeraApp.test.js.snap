// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primera App test 1`] = `
<Fragment>
  <h1>
    KALADIN
  </h1>
  <p>
    Relleno
  </p>
</Fragment>
`;
